.gitignore

```diff
@@ -11,8 +16,200 @@ HELP.md
 .springBeans
 .sts4-cache
 
-### IntelliJ IDEA ###
-**/.idea
-**/*.iws
-**/*.iml
-**/*.ipr
+# Created by https://www.gitignore.io/api/java,linux,macos,maven,windows,intellij+all
+# Edit at https://www.gitignore.io/?templates=java,linux,macos,maven,windows,intellij+all
+
+### Intellij+all ###
+# Covers JetBrains IDEs: IntelliJ, RubyMine, PhpStorm, AppCode, PyCharm, CLion, Android Studio and WebStorm
+# Reference: https://intellij-support.jetbrains.com/hc/en-us/articles/206544839
+
+# User-specific stuff
+.idea/**/workspace.xml
+.idea/**/tasks.xml
+.idea/**/usage.statistics.xml
+.idea/**/dictionaries
+.idea/**/shelf
+
+# Generated files
+.idea/**/contentModel.xml
+
+# Sensitive or high-churn files
+.idea/**/dataSources/
```

Comment:
all .idea/ lines can be deleted, full Intellij folder use ignored below

Reply:
Yep. But I don't like modifying the generated section of the gitignore, because I'll likely be overridden in the future. If you want to remove this, I can just remove the IntelliJ rules from the generated section and move them above. But I personally won't bother.